jax/random.py

```diff
@@ -412,14 +416,14 @@ def bernoulli(key, p=onp.float32(0.5), shape=()):
 def _bernoulli(key, p, shape):
   _check_shape("bernoulli", shape)
   shape = shape or onp.shape(p)
-  if not onp.issubdtype(onp.float32, lax.dtype(p)):
-    p = lax.convert_element_type(p, onp.float32)
+  if not onp.issubdtype(onp.float64, lax.dtype(p)):
+    p = lax.convert_element_type(p, onp.float64)
```

Comment:
I think it is better to have `p=0.5` by default and do `p = lax.convert_element_type(p, dtype)` here with `dtype = xla_bridge.canonicalize_dtype(lax.dtype(p))`

Reply:
I revised the logic a bit; PTAL and let me know what you think.

I think it should be an error if the `p` parameter is anything but a subdtype of `onp.floating`, so now we raise the error instead. I put the error-raising and casting outside the `@jit` subroutine for the weak reason that canonicalizing outside could get us more cache hits.

Reply:
Yeah, the change looks great!

Reply:
Thanks for finding this, and checking it!